### PR TITLE
Removed flags from data draft

### PR DESCRIPTION
**Pull request by talmi**
Removed the loopback flag, immediate export flag, and active flag.
Immediate export will be replaced by a dedicated IOAM-Type in the next version.
Active flag will be in a new draft.

### CODE REVIEW

**Comment by barakgafni** [comment]
Re 2: I believe we should keep it there at the moment with TBD reference to the new option, just so we will not miss it. Anyway 06 is not for actual use.

**Comment by talmi** [comment]
Per Barak's suggestion, I added a "TBD" to the two places where immediate export is mentioned. This can be updated once we add the new IOAM type.

File: drafts/draft-ietf-ippm-ioam-data.xml [hunk 1]
```diff
@@ -805,96 +805,18 @@ The trace option data MUST be 4-octet aligned:
...
-              <t anchor="TraceFlags" hangText="Flags">4-bit field. The
-              following flags are defined: <list style="hanging">
+              <t anchor="TraceFlags" hangText="Flags">4-bit field. Flags
+			  are allocated by IANA, as specified in 
+			  <xref target="Flags-Registry-Sec"></xref>.
+			  The current document allocates a single flag as follows: 
+			  <list style="hanging">
                   <t hangText="Bit 0">"Overflow" (O-bit) (most significant
                   bit). This bit is set by the network element if there is not
                   enough number of octets left to record node data, no field
```
Comment (brockners) [inline drafts/draft-ietf-ippm-ioam-data.xml]: s/if there is not enough number of octets left/if there are not enough octets left/

File: drafts/draft-ietf-ippm-ioam-data.xml [hunk 2]
```diff
@@ -1915,22 +1837,16 @@ IOAM proof of transit option of IOAM POT Type 0:
         via RFC Required process as per <xref target="RFC8126"></xref>.</t>
       </section>
 
-      <section title="IOAM Trace Flags Registry">
+      <section anchor="Flags-Registry-Sec" title="IOAM Trace Flags Registry">
         <t>This registry defines code points for each bit in the 4 bit flags
         for Pre-allocated trace option and Incremental trace option defined in
```
Comment (brockners) [inline drafts/draft-ietf-ippm-ioam-data.xml]: s/flags for Pre-allocated trace option and Incremental trace option/for the Pre-allocated trace option and for the Incremental trace option/

File: drafts/draft-ietf-ippm-ioam-data.xml [hunk 3]
```diff
@@ -2010,10 +1926,11 @@ IOAM proof of transit option of IOAM POT Type 0:
       <t>The data elements of IOAM can be used for network reconnaissance,
       allowing attackers to collect information about network paths,
       performance, queue states, buffer occupancy and other information. Note
-      that in case IOAM is used in "immediate export" mode, the IOAM related
-      trace information would not be available in the customer data packets,
-      but would be exported by every IOAM node. IOAM data export and securing
-      IOAM data export is outside the scope of this document.</t>
+      that in case IOAM is used in "immediate export" mode (TBD - to be 
+	  updated), the IOAM related trace information would not be available in
```
Comment (brockners) [inline drafts/draft-ietf-ippm-ioam-data.xml]: rather than say "TBD" - should we add something like "reference to be added in a future revision"?

Comment (brockners) [inline drafts/draft-ietf-ippm-ioam-data.xml]: Could we be specific about what "information" would not be available. E.g.
s/information/information (i.e. node data[i])/

File: drafts/draft-ietf-ippm-ioam-data.xml [hunk 4]
```diff
@@ -2010,10 +1926,11 @@ IOAM proof of transit option of IOAM POT Type 0:
       <t>The data elements of IOAM can be used for network reconnaissance,
       allowing attackers to collect information about network paths,
       performance, queue states, buffer occupancy and other information. Note
-      that in case IOAM is used in "immediate export" mode, the IOAM related
-      trace information would not be available in the customer data packets,
-      but would be exported by every IOAM node. IOAM data export and securing
-      IOAM data export is outside the scope of this document.</t>
+      that in case IOAM is used in "immediate export" mode (TBD - to be 
+	  updated), the IOAM related trace information would not be available in 
+	  the customer data packets, but would be exported by every IOAM node.
```
Comment (brockners) [inline drafts/draft-ietf-ippm-ioam-data.xml]: What immediate export is going to export is a TBD for now. Let's reflect this in the text - e.g.

s/but would be exported by every IOAM node/but would trigger export of  packet related OAM information at every node/.

File: drafts/draft-ietf-ippm-ioam-data.xml [hunk 5]
```diff
@@ -2041,8 +1958,9 @@ IOAM proof of transit option of IOAM POT Type 0:
       current network domain the network operator is expected to enforce
       policies that prevent IOAM traffic from leaking outside of the IOAM
       domain, and prevent IOAM data from outside the domain to be processed
-      and used within the domain. Another way to prevent the data to get
-      leaked is using the Immediate Export mode of the trace option.</t>
+      and used within the domain. Note that the Immediate Export mode (TBD
+	  - to be updated) can mitigate the potential threat of IOAM data leaking
```
Comment (brockners) [inline drafts/draft-ietf-ippm-ioam-data.xml]: see above  - add "need reference rather than TBD".

**Comment by brockners** [comment]
IMHO we're good to go and can merge the patch. 

@mickeyspiegel : IMHO the behavior of "IOAM may also be carried out on cloned or sampled copies of data packets, when the operator prefers not to directly modify data packets for IOAM purposes." could be achieved without the "active" flag: As long as you have some means to identify the clone, then you should be good. This does not have to be the active flag - but you could (deployment dependent) steal the bit somewhere else in the header.

**Pull request merged**